### PR TITLE
fix:(2259)update client secret table

### LINKED ISSUE #2259: update to client secrets table
Page:
- https://site-local.fusionauth.io/docs/v1/tech/oauth/endpoints#client-secret-table

The table used to describe when the `client_secret` is required isn't quite correct. 

1. Instead of calling out where the `client_secret` is required - body or header, we should just say it is required. The `client_secret` can be provided in the Basic auth header, or in the request body. 
2. You can always send PKCE, so instead of saying `N/A` shouldn't we say `yes` or `no` indicating if it is used by the caller?

### PR DESCRIPTION
**Pull request by alex-fusionauth**
Fixes #2259 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205076385141046

### CODE REVIEW

**Comment by alex-fusionauth** [comment]
@lyleschemmerling can we use markdown tables and adjust the CSS?

**Comment by lyleschemmerling** [comment]
@alex-fusionauth the way to adjust it would be by adding classes for `prose:thead` and `prose:tr` on line 98 in `Default.astro` but that does control styling on all tables

**Comment by alex-fusionauth** [comment]
Closing, @lyleschemmerling's update fixed

**Comment by alex-fusionauth** [comment]
@lyleschemmerling can you review once more?

File: astro/src/content/docs/lifecycle/authenticate-users/oauth/endpoints.mdx [hunk 1]
```diff
@@ -1962,21 +1963,16 @@ For versions `1.28.0` or later, use <InlineField>oauthConfiguration.clientAuthen
...
+| Authorization Code Grant   | `Required`                       | -         | Yes                       |
+| Authorization Code Grant   | `NotRequired`                    | -         | No                        |
+| Authorization Code Grant   | `NotRequiredWhenUsingPKCE`       | Yes       | No                        |
+| Authorization Code Grant   | `NotRequiredWhenUsingPKCE`       | No        | Yes                       |
+| Implicit Grant             | N/A                              | -         | -                         |
+| Password Credentials Grant | `Required`                       | -         | Yes                       |
+| Password Credentials Grant | `NotRequired`                    | -         | No                        |
+| Refresh Token Grant        | `Required`                       | -         | Yes                       |
+| Refresh Token Grant        | `NotRequired`                    | -         | No                        |
+|                            |                                  |           |                           |
```
Comment (lyleschemmerling) [inline astro/src/content/docs/lifecycle/authenticate-users/oauth/endpoints.mdx:1978]: more apparent now with the new styles but we should drop the bottom empty row

**Review by lyleschemmerling — APPROVED** [review]

**Pull request merged**